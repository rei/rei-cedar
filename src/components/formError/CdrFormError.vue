<template>
  <div
    :class="style[baseClass]"
    role="status"
  >
    <span :class="style[iconClass]">
      <icon-error-stroke
        size="small"
        inherit-color
      />
    </span>
    <slot name="error">
      <span>{{ error }}</span>
    </slot>
  </div>
</template>

<script>
import { defineComponent, useCssModule } from 'vue';
import IconErrorStroke from '../icon/comps/error-stroke.vue';

export default defineComponent({
  name: 'CdrFormError',
  components: { IconErrorStroke },
  props: {
    error: [Boolean, String],
  },

  setup() {
    return {
      style: useCssModule(),
      baseClass: 'cdr-form-error',
      iconClass: 'cdr-form-error__icon',
    };
  },
});
</script>

<style lang="scss" module src="./styles/CdrFormError.module.scss">
</style>
