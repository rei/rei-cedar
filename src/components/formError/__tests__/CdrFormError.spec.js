import { mount } from '../../../../test/vue-jest-style-workaround.js';
import CdrFormError from '../CdrFormError.vue';

describe('CdrFormError', () => {
  describe('using just the error prop (no slot)', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(CdrFormError, {
        propsData: {
          error: 'yahoo!',
        },
      });
    })

    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders error prop', () => {
      expect(wrapper.find('.cdr-form-error').text()).toBe('yahoo!');
    });
  })

  describe('using an error slot', () => {
    let wrapper;
    beforeEach(() => {
      wrapper =  mount(CdrFormError, {
        propsData: {
          error: true
        },
        slots: {
          'error': 'whoops',
        },
      });
    })

    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders error slot', () => {
      expect(wrapper.find('.cdr-form-error').text()).toBe('whoops');
    });
  });
});
