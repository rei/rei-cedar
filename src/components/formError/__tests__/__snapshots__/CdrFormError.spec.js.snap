// Vitest Snapshot v1

exports[`CdrFormError > using an error slot > renders correctly 1`] = `
<div
  class="cdr-form-error"
  role="status"
>
  <span
    class="cdr-form-error__icon"
  >
    <svg
      aria-hidden="true"
      class="cdr-icon cdr-icon--small cdr-icon--inherit-color"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      
      
      
      <g>
        <path
          d="M15.293 16.707a1 1 0 001.414-1.414L9.121 7.707a1 1 0 10-1.414 1.414l7.586 7.586z"
          role="presentation"
        />
        <path
          clip-rule="evenodd"
          d="M22 12c0 5.523-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2s10 4.477 10 10zm-2 0a8 8 0 11-16 0 8 8 0 0116 0z"
          fill-rule="evenodd"
        />
      </g>
      
      <!--v-if-->
    </svg>
  </span>
  
  whoops
  
</div>
`;

exports[`CdrFormError > using just the error prop (no slot) > renders correctly 1`] = `
<div
  class="cdr-form-error"
  role="status"
>
  <span
    class="cdr-form-error__icon"
  >
    <svg
      aria-hidden="true"
      class="cdr-icon cdr-icon--small cdr-icon--inherit-color"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      
      
      
      <g>
        <path
          d="M15.293 16.707a1 1 0 001.414-1.414L9.121 7.707a1 1 0 10-1.414 1.414l7.586 7.586z"
          role="presentation"
        />
        <path
          clip-rule="evenodd"
          d="M22 12c0 5.523-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2s10 4.477 10 10zm-2 0a8 8 0 11-16 0 8 8 0 0116 0z"
          fill-rule="evenodd"
        />
      </g>
      
      <!--v-if-->
    </svg>
  </span>
  
  <span>
    yahoo!
  </span>
  
</div>
`;
